grpc/pb-marshalling.go

```diff
@@ -132,15 +134,20 @@ func pbToChallenge(in *corepb.Challenge) (challenge core.Challenge, err error) {
 	if err != nil {
 		return core.Challenge{}, err
 	}
-	return core.Challenge{
-		ID:                       *in.Id,
-		Type:                     *in.Type,
-		Status:                   core.AcmeStatus(*in.Status),
-		Token:                    *in.Token,
-		ProvidedKeyAuthorization: *in.KeyAuthorization,
-		Error:                    prob,
-		ValidationRecord:         recordAry,
-	}, nil
+	ch := core.Challenge{
+		Type:             *in.Type,
+		Status:           core.AcmeStatus(*in.Status),
+		Token:            *in.Token,
+		Error:            prob,
+		ValidationRecord: recordAry,
+	}
+	if in.Id != nil {
+		ch.ID = *in.Id
+	}
+	if in.KeyAuthorization != nil {
+		ch.ProvidedKeyAuthorization = *in.KeyAuthorization
```

Comment:
We really need to rename this field :sweat: (_Probably better for a separate PR_)

Reply:
Yeah, I think given the already large scope here it makes sense to bump this for another time...